app/src/main/java/com/orange/ods/app/ui/components/banners/ComponentBanners.kt

```diff
@@ -72,35 +77,61 @@ fun ComponentBanners() {
+                val onButton1ClickText = stringResource(id = R.string.component_element_button1)
+                val onButton1Click = { clickOnElement(context, onButton1ClickText) }
+                val button2Text = if (hasButton2) stringResource(id = R.string.component_banner_detail) else null
+                val onButton2ClickText = stringResource(id = R.string.component_element_button2)
+                val onButton2Click = { clickOnElement(context, onButton2ClickText) }
+                UiFramework<OdsBannerBinding>(
+                    compose = {
+                        OdsBanner(
+                            message = message,
+                            button1Text = button1Text,
+                            button2Text = button2Text,
+                            image = if (hasIcon) rememberAsyncImagePainter(
+                                model = recipe.imageUrl,
+                                placeholder = painterResource(id = R.drawable.placeholder),
+                                error = painterResource(id = R.drawable.placeholder)
+                            ) else null,
+                            onButton1Click = onButton1Click,
+                            onButton2Click = onButton2Click,
+                        )
 
-                CodeImplementationColumn {
-                    CommonTechnicalTextColumn(
-                        componentName = OdsComponent.OdsBanner.name
-                    ) {
-                        if (hasTwoTextLines) TechnicalText(text = " message = \"${recipe.description}\"")
-                        else TechnicalText(text = " message = \"${recipe.title}\"")
-                        if (hasIcon) TechnicalText(text = " image = painterResource(id = R.drawable.image)")
-                        TechnicalText(text = " button1Text = \"${stringResource(id = R.string.component_banner_dismiss)}\"")
-                        if (hasButton2) TechnicalText(text = " button2Text = \"${stringResource(id = R.string.component_banner_detail)}\"")
+                        CodeImplementationColumn {
+                            CommonTechnicalTextColumn(
+                                componentName = OdsComponent.OdsBanner.name
+                            ) {
+                                if (hasTwoTextLines) TechnicalText(text = " message = \"${recipe.description}\"")
+                                else TechnicalText(text = " message = \"${recipe.title}\"")
+                                if (hasIcon) TechnicalText(text = " image = painterResource(id = R.drawable.image)")
+                                TechnicalText(text = " button1Text = \"${stringResource(id = R.string.component_banner_dismiss)}\"")
+                                if (hasButton2) TechnicalText(text = " button2Text = \"${stringResource(id = R.string.component_banner_detail)}\"")
+                            }
+                        }
+                    },
+                    xml = {
```

Comment:
To be complete, we also have to display code implementation for XML and to add XML components documentation. Maybe create another issue for that?

Reply:
Yes, it's probably a lot of work, so I thought we could do this in another PR :+1:

Reply:
I'm going to create a new issue for that